universal-application-tool-0.0.1/app/services/export/CsvExporter.java

```diff
@@ -65,9 +75,35 @@ public void export(Application application, Writer writer) throws IOException {
         case SUBMITTER_EMAIL:
           printer.print(application.getSubmitterEmail().orElse("Applicant"));
           break;
+        case OPAQUE_ID:
+          printer.print(opaqueIdentifier(this.secret.get(), application.getApplicant().id));
+          break;
+        case APPLICANT_OPAQUE:
+          Optional<String> applicantValue =
+              application.getApplicantData().readAsString(column.jsonPath().orElseThrow());
+          // We still hash the empty value.
+          printer.print(opaqueIdentifier(this.secret.get(), applicantValue.orElse(EMPTY_VALUE)));
```

Comment:
Question (non-blocking): should this throw if the secret is missing?

Reply:
Yep, and it does - I could make the error a little better, will do.